lib/connection.cpp

```diff
@@ -737,29 +738,36 @@ ForgetRoomJob* Connection::forgetRoom(const QString& id)
     if (room && room->joinState() != JoinState::Leave)
     {
         auto leaveJob = room->leaveRoom();
-        connect(leaveJob, &BaseJob::success, this, [this, forgetJob, room] {
-            forgetJob->start(connectionData());
-            // If the matching /sync response hasn't arrived yet, mark the room
-            // for explicit deletion
-            if (room->joinState() != JoinState::Leave)
-                d->roomIdsToForget.push_back(room->id());
+        connect(leaveJob, &BaseJob::result, this, [this, leaveJob, forgetJob, room] {
+            // After leave, continue if there is no error or the room id is not found (IncorrectRequestError)
+            if(!leaveJob->error() || leaveJob->error() == BaseJob::StatusCode::IncorrectRequestError || leaveJob->error() == BaseJob::StatusCode::UnknownError) {
```

Comment:
Please don't ignore `IncorrectRequestError` - you can't imagine how many times this code helped me debug the code already.
```suggestion
            if(!leaveJob->error() ||
                    leaveJob->error() == BaseJob::StatusCode::UnknownError) {
```

Reply:
Removed IncorrectRequestError in next PR